DOC-10303 release/7.2: Adds URL functions to urlencode and urldecode (#2927)
couchbase/docs-server: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc

}
----
====

> [#fn-str-upper]

Reviewer: Each function needs to have a unique anchor:

```suggestion
[#fn-str-urlencode]
```

== URLENCODE(plain_string)

=== Description

> Returns the input string encoded for use in a URL

Reviewer: Full stop at end

```suggestion
Returns the input string encoded for use in a URL.
```

Author: I did wander about that full stop, I just copied what was there. Lesson for me — follow my intuition  💪

=== Arguments

> in_str:: A string, or any valid xref:n1ql-language-reference/index.adoc[expression] which evaluates to a string, that is the string to convert to upper case.

Reviewer: Needs to be consistent with syntax given in the function title. Also fix description:

```suggestion
plain_string:: A string, or any valid xref:n1ql-language-reference/index.adoc[expression] which evaluates to a string, that is the string to URL-encode.
```

=== Return Value

> A string representing the input string converted to upper case.

Reviewer: Fix description. Also we need to cover the common edge cases &mdash; although I admit this page is not very good at that!

```suggestion
The input string, with any reserved characters replaced by percent encoding.

If the input argument is MISSING, returns MISSING.
If the input argument is non-string, returns NULL.
```

=== Example

> ====
> [source,text]
> ----
> SELECT%20name%20FROM%20%60travel-sample%60.inventory.hotel%20LIMIT%201%3B
> ----
> 
> ====

Reviewer: Okay, you've shown the _result_ of the function here, but we need to see a complete example, with input and output.

```suggestion
====
[source,sqlpp]
----
SELECT URLENCODE("SELECT name FROM `travel-sample`.inventory.hotel LIMIT 1;") AS encoded;
----

[source,json]
----
[
  {
    "encoded": SELECT%20name%20FROM%20%60travel-sample%60.inventory.hotel%20LIMIT%201%3B
  }
]
----
====
```

> [#fn-str-upper]

Reviewer: Each function needs to have a unique anchor:

```suggestion
[#fn-str-urldecode]
```

== URLDECODE(encoded_string)

=== Description

> Decodes the URL-encoded input string and returns the resulting plain string

Reviewer: Full stop at end:

```suggestion
Decodes the URL-encoded input string and returns the resulting plain string.
```

=== Arguments

> in_str:: A string, or any valid xref:n1ql-language-reference/index.adoc[expression] which evaluates to a string, that is the string to convert to upper case.

Reviewer: Needs to be consistent with syntax given in the function title. Also fix description:

```suggestion
encoded_string:: A string, or any valid xref:n1ql-language-reference/index.adoc[expression] which evaluates to a string, that is the string to URL-decode.
```

=== Return Value

> A string representing the input string converted to upper case.

Reviewer: Again, fix description, and cover the common edge cases:

```suggestion
The input string, with any percent encoding replaced by reserved characters.

If the input argument is MISSING, returns MISSING.
If the input argument is non-string, or if the argument is a string containing a single percent character `%`, returns NULL.
```

=== Example

> ====
> [source,text]
> ----
> SELECT%20name%20FROM%20%60travel-sample%60.inventory.hotel%20LIMIT%201%3B”) AS decode;
> SELECT name FROM travel-sample.inventory.hotel LIMIT 1;
> ----
> ====

Reviewer: Again, need to show complete example, with input and output.

```suggestion
====
[source,sqlpp]
----
SELECT URLDECODE("SELECT%20name%20FROM%20%60travel-sample%60.inventory.hotel%20LIMIT%201%3B") AS decoded;
----

[source,json]
----
[
  {
    "decoded": "SELECT name FROM `travel-sample`.inventory.hotel LIMIT 1;"
  }
]
----
====
```
